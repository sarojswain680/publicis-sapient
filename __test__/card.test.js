import '@testing-library/jest-dom/extend-expect';
import { cleanup } from '@testing-library/react';
import Enzyme, { shallow } from 'enzyme';
import renderer from "react-test-renderer";
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import Card from '../src/components/Modules/card';

Enzyme.configure({ adapter: new Adapter() })

afterEach(cleanup)


it("renderers card without crashing", () => {

    const data = {
        links: {
            mission_patch: "https://images2.imgbox.com/40/e3/GypSkayF_o.png",
            article_link: "https://www.space.com/2196-spacex-inaugural-falcon-1-rocket-lost-launch.html",
            video_link: "https://www.youtube.com/watch?v=0a_00nJ_Y88",
        },
        mission_name: "FalconSat",
        flight_number: 1,
        launch_success: false,
        mission_id: "EEBBJUEE",
        launch_site: {
            launch_year: "2006",
        },
        rocket: { rocket_type: "Merlin A" }
    }

    const wrapper = shallow(<Card data={data} />);
    const root = wrapper.find('#text')
    expect(root.first().text()).toEqual('Mission Ids:- EEBBJUEE')
})

it("matches snapshot", () => {
    const data = {
        links: {
            mission_patch: "https://images2.imgbox.com/40/e3/GypSkayF_o.png",
            article_link: "https://www.space.com/2196-spacex-inaugural-falcon-1-rocket-lost-launch.html",
            video_link: "https://www.youtube.com/watch?v=0a_00nJ_Y88",
        },
        mission_name: "FalconSat",
        flight_number: 1,
        launch_success: false,
        mission_id: "EEBBJUEE",
        launch_site: {
            launch_year: "2006",
        },
        rocket: { rocket_type: "Merlin A" }
    }
    const tree = renderer.create(<Card data={data}></Card>).toJSON();

    expect(tree).toMatchSnapshot();
})
