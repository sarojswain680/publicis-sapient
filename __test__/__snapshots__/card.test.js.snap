// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  data-testid="button"
  id="card"
>
  <div
    className="MuiCardMedia-root makeStyles-media-2"
    style={
      Object {
        "backgroundImage": "url(\\"https://images2.imgbox.com/40/e3/GypSkayF_o.png\\")",
      }
    }
    title="FalconSat"
  />
  <div
    className="MuiCardHeader-root sub-text"
  >
    <div
      className="MuiCardHeader-avatar"
    >
      <div
        aria-label="mission-name"
        className="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-5 MuiAvatar-colorDefault"
      >
        F
      </div>
    </div>
    <div
      className="MuiCardHeader-content"
    >
      <span
        className="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
      >
        FalconSat# 1
      </span>
    </div>
  </div>
  <div
    className="MuiCardActions-root"
  >
    <button
      aria-label="share"
      className="MuiButtonBase-root MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <a
          href="https://www.space.com/2196-spacex-inaugural-falcon-1-rocket-lost-launch.html"
          target="_blank"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm-5 14H4v-4h11v4zm0-5H4V9h11v4zm5 5h-4V9h4v9z"
            />
          </svg>
        </a>
      </span>
    </button>
    <button
      aria-label="share"
      className="MuiButtonBase-root MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <a
          href="https://www.youtube.com/watch?v=0a_00nJ_Y88"
          target="_blank"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 15l5.19-3L10 9v6m11.56-7.83c.13.47.22 1.1.28 1.9.07.8.1 1.49.1 2.09L22 12c0 2.19-.16 3.8-.44 4.83-.25.9-.83 1.48-1.73 1.73-.47.13-1.33.22-2.65.28-1.3.07-2.49.1-3.59.1L12 19c-4.19 0-6.8-.16-7.83-.44-.9-.25-1.48-.83-1.73-1.73-.13-.47-.22-1.1-.28-1.9-.07-.8-.1-1.49-.1-2.09L2 12c0-2.19.16-3.8.44-4.83.25-.9.83-1.48 1.73-1.73.47-.13 1.33-.22 2.65-.28 1.3-.07 2.49-.1 3.59-.1L12 5c4.19 0 6.8.16 7.83.44.9.25 1.48.83 1.73 1.73z"
            />
          </svg>
        </a>
      </span>
    </button>
  </div>
  <div
    className="MuiCardContent-root"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph"
      id="text"
    >
      Mission Ids:- EEBBJUEE
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph"
    >
      Launch Year:- 2006
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph"
    >
      Successful Launch:- false
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph"
    >
      Rocket Type:- Merlin A
    </p>
  </div>
</div>
`;
